### ISSUE TITLE
Feature request: bookdown.skeleton function

### ISSUE THREAD

**Issue by mingsnu** [rstudio/bookdown#225]
I think we need a `bookdown.skeleton` function to create a skeleton for a new book. I would be pleased to contribute to this if necessary. Thank you!

**Comment by yihui** [comment]
Similar issue as #116. We will add this feature in the RStudio IDE, and you will be able to create a book project from the menu. For now, you can get started with the bookdown-demo project: https://github.com/rstudio/bookdown-demo

**Comment by mingsnu** [comment]
It would be great to add this feature to RStudio IDE, then I will wait. Cloning the `bookdown-demo` or copying from my own template is what am I doing right now. Thank you:)

**Comment by yihui** [comment]
Actually I think such a function makes sense, and the RStudio IDE can just call such a function to create a book skeleton.

**Comment by yihui** [comment]
You can create a skeleton book project using the RStudio IDE and the development version of bookdown now: https://rstudio.github.io/rstudio-extensions/rstudio_project_templates.html

**Comment by ColinFay** [comment]
Hi @yihui 

I know this issue is closed, yet I still think this could be a nice feature to be able to add a bookdown skeleton to your current directory. 

Here are two use case I've been looking for this feature: 
- Creating a package which includes a bookdown 
- Writing an analysis, which was first supposed to be a plain html, but that I wanted after to turn into a bookdown. 

I know this is just copy and pasting in both case, yet in both these cases it would be more secure to add a skeleton in the current dir. 

This seems like a pretty straightforward thing to code : 
-> Adding current bookdown templates in inst/ 
-> code bookdown::use_bookdown() which would fetch and copy these template to the current directory. 

If you think this is a good idea, I'll be glad to send you a PR with these elements.

**Comment by yihui** [comment]
@ColinFay Have you checked the functions `bookdown::book_skeleton` and `bookdown:::bookdown_skeleton` (N.B. `:::`)? The RStudio IDE uses the latter, and you can use it, too, e.g., `bookdown:::bookdown_skeleton(getwd())`.

**Comment by krlmlr** [comment]
Did you mean to export `book_skeleton()`, and perhaps enalbe generation of `.Rd` documentation for this function? I was looking for exported functions too.

**Comment by krlmlr** [comment]
The initial comment for that function reads that it's not meant to be exported, maybe you could consider this? It seems surprisingly difficult to start a bookdown project when offline -- a link to that function could be featured prominently on `?bookdown::bookdown` .

**Comment by yihui** [comment]
Since RStudio 1.2 has been released, we should update the book section https://bookdown.org/yihui/bookdown/get-started.html and recommend users to create book projects from the RStudio IDE just like website projects: https://bookdown.org/yihui/blogdown/rstudio-ide.html (cc @angela-li)

**Comment by krlmlr** [comment]
Do you consider the case when your users can't access https://bookdown.org just when they want to start a project, but have everything else set up otherwise? Or when they don't have an installation of the RStudio IDE on their current infrastructure?

**Comment by angela-li** [comment]
These are really good points! I think there is a case to be made for something like a `create_book()` function that wraps `bookdown_skeleton(getwd())`, so that people don't have to dig into the internals or [Advanced R Markdown tutorials](https://arm.rbind.io/slides/bookdown.html#10) to find `bookdown:::bookdown_skeleton(getwd())` (thanks @apreshill!) which can be challenging syntax for beginners. Questions provoked by this include: _what is :::? what is namespacing? what are internal functions? why is getwd() necessary?_, which is a lot if you're just trying to set up your first book 😄 

Also, as @krlmlr mentioned, not everyone is using the IDE, and even those who do (me!) forget that you can create book projects through the IDE. And making the book through the IDE doesn't work well with [the workflow suggested in Happy Git with R](https://happygitwithr.com/new-github-first.html) where you make a repo on Github, clone it as a new version controlled R project, and just need to set up the book structure.

**tl;dr** Making a function like `create_book()` could be very helpful for someone getting started with the package, in terms of making functions discoverable, and it would lower the barrier to getting started with bookdown, encouraging more books to be written! 🎉📚

**Comment by yihui** [comment]
When I "cc @angela-li" above, I meant "Could you help?" :) Since we have talked about this problem back in April, and I absolutely agreed with you, I'm just waiting for pull requests 😉

**Comment by angela-li** [comment]
Oh, great! Thank you for the reminder and encouragement. I will take a look at this and hope to get a PR submitted!

**Comment by cderv** [comment]
Hi everyone, thanks for the wait on this one ! 

**bookdown** will now have two functions for create a template if you don't want to use the IDE "New Project" feature
* `create_gitbook()`
* `create_bs4_book()`

Choose the one you want for the HTML book format which has your preference.

Help page: https://pkgs.rstudio.com/bookdown/reference/create_book.html

We welcome any feedback, and any PR to improve the included template. Feel free to reach out! 

Thanks.
